src/Umbraco.Core/IO/IOHelper.cs

```diff
@@ -13,7 +13,9 @@
 namespace Umbraco.Core.IO
 {
 	public static class IOHelper
-    {
+	{
+	    public static bool IAmUnitTestingSoNeverUseHttpContextEver = false;

```

Comment:
lol! ... ok fair enough :) Out of curiosity how come the null check on HttpContext.Current doesn't work for your test scenario and you need this flag?